### DISCUSSION TITLE
CoreWCF vs gRPC

### DISCUSSION THREAD

**Discussion by CD1010** [CoreWCF/CoreWCF#789] [Q&A]
We have a legacy client server app that uses WCF for all communications between client and server.
The client is a desktop app that has heavy dependency on 3rd party controls that require .net Framework.  Moving it to .net core would likely break those components.

We would like to at least move the server to .net core and host on linux, but upon researching gRPC, i found that because of .net 4.8, 
we are stuck with http/1.1 as the transport from the client.  I understand that CoreWCF can support a 4.8 client.

- Would we get full bidirectional transport support with a .net 4.8 client and core server if we went to CoreWCF?
- We currently have issues with cross domain security ( corrs ) with the current implementation of WCF.  Does CORE WCF do anything to solve this type of problem?

**Comment by mconnew** [comment]
CoreWCF does support Duplex communication, depending on the binding (same as WCF). Can you provide your binding that you are currently using and I can tell you if what you currently have is fully supported, or if there's some changes you can make to give you equivalent functionality.  
Can you explain the problems you are having with CORS? It's a browser spec which allows enforcing or relaxing cross domain content. I.e. if you visit the website www.example.com and the policy is same origin and some component on the website is referencing an image from www.example.org, then the browser will refuse to retrieve the image. With a web service requests, you make a request to a service hosted at example.com, and it replies with the response and that's the end of it. There is no automatic downloading of resources from another domain based on the content of the response.

**Reply by CD1010** [reply]
Well, not exactly CORS in browser sense.  This is a fat client app on windows uses WCF client code to communicate with a central server.  they are using NetTCPBindings.  Transport speed is very important since app is very chatty.
They have to work around it by using certs, but that is painful.  The way is was described is end users are logged into domains that are different than the domain under which the server runs.
